### PR TITLE
Fix tests and change some objects

### PR DESCRIPTION
**Pull request by Lash-L**
Changes:
- Removed async_setup - when configuration setup isn't done in yaml, it doesn't need to exist.
- Changed to OptionsFlowwithConfigEntry
- Created two fixtures to get either a mockconfigentry or a mock config entry that has been set up
- Changed how setup of entries works
- Instead of directly calling functions in init, we now use hass to call them

I will have to change either this Pr or the other depending on which gets merged first

### CODE REVIEW

**Comment by agittins** [comment]
I figured this one might be easier to rebase, so I merged the formatting PR first.

File: custom_components/bermuda/config_flow.py [hunk 1]
```diff
@@ -104,14 +105,12 @@ def async_get_options_flow(config_entry):
...
-class BermudaOptionsFlowHandler(config_entries.OptionsFlow):
+class BermudaOptionsFlowHandler(OptionsFlowWithConfigEntry):
     """Config flow options handler for bermuda."""
 
     def __init__(self, config_entry: ConfigEntry) -> None:
         """Initialize HACS options flow."""
-        super().__init__()
-        self.config_entry = config_entry
-        self.options = dict(config_entry.options)
+        super().__init__(config_entry)
```
Comment (agittins) [inline custom_components/bermuda/config_flow.py:113]: Looks like the super will save the config_entry and options into `self._config_entry` and `self._options`, so all our references will need to be updated to use the `_` prefix on those.

File: tests/test_config_flow.py [hunk 2]
```diff
@@ -19,24 +17,9 @@
 from .const import MOCK_CONFIG
 from .const import MOCK_OPTIONS_GLOBALS
 
-
 # This fixture bypasses the actual setup of the integration
```
Comment (agittins) [inline tests/test_config_flow.py:20]: I reckon delete the comment, too.

**Review by agittins — COMMENTED** [review]
This is great, thanks! I was pretty baffled with the whole testing thing :-)

File: custom_components/bermuda/config_flow.py (hunk 1 above)
Comment (Lash-L) [reply custom_components/bermuda/config_flow.py:113]: image
They set .options and .config_entry as options

Comment (agittins) [reply custom_components/bermuda/config_flow.py:113]: SMH! I didn't see the property methods, my bad! 🤦🏼

**Review by agittins — APPROVED** [review]
Awesome!

**Pull request merged**